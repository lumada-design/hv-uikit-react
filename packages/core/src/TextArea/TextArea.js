/*
 * Copyright 2019 Hitachi Vantara Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react";
import PropTypes from "prop-types";
import isNil from "lodash/isNil";
import classNames from "classnames";
import HvTypography from "../Typography";
import Input from "../Input";

/**
 * A text area component wrapping the input box, it allows the input of paragraph of text.
 * alongside this it can provide a validation for the max character quantity
 *
 * @class HvTextArea
 * @extends {React.Component}
 */
class HvTextArea extends React.Component {
  constructor(props) {
    super(props);
    const { value } = this.props;
    this.state = {
      currentValueLength: value.length
    };
  }

  /**
   * Updates the length of the string while is being inputted, also executes the user onChange
   * allowing the customization of the input if required.
   *
   * @param {String} value - The value provided by the HvInput
   */
  onChangeHandler = value => {
    const { onChange, maxCharQuantity } = this.props;
    const newValue = onChange(value);

    const textAreaValue =
      isNil(maxCharQuantity) || newValue.length < maxCharQuantity
        ? newValue
        : newValue.substring(0, maxCharQuantity);

    this.setState({
      currentValueLength: textAreaValue.length
    });

    return textAreaValue;
  };

  render() {
    const {
      classes,
      inputTextConfiguration,
      maxCharQuantity,
      rows,
      value,
      disabled
    } = this.props;

    const { currentValueLength } = this.state;

    return (
      <>
        <Input
          classes={{
            container: classes.container,
            input: classes.input
          }}
          inputTextConfiguration={inputTextConfiguration}
          value={value}
          onChange={this.onChangeHandler}
          multiline
          rows={rows}
          disabled={disabled}
          validate={false}
          iconVisible={false}
        />
        {maxCharQuantity ? (
          <div className={classes.characterCounter}>
            <HvTypography
              className={classNames(classes.inline, {
                [classes.currentCounter]: !disabled,
                [classes.disabled]: disabled
              })}
              variant="labelText"
            >
              {currentValueLength}
            </HvTypography>
            <HvTypography
              className={classNames(classes.inline, classes.separator, {
                [classes.maxCharacter]: !disabled,
                [classes.disabled]: disabled
              })}
              variant="infoText"
            >
              /
            </HvTypography>
            <HvTypography
              className={classNames(classes.inline, {
                [classes.maxCharacter]: !disabled,
                [classes.disabled]: disabled
              })}
              variant="infoText"
            >
              {maxCharQuantity}
            </HvTypography>
          </div>
        ) : null}
      </>
    );
  }
}

// [classes.currentCounter]:!disabled,[classes.disabled]:disabled

HvTextArea.propTypes = {
  /**
   *  Styles applied to the Drawer Paper element.
   */
  classes: PropTypes.PropTypes.shape({
    /**
     * Style applied on the text area input box.
     */
    input: PropTypes.string,
    /**
     * Style applied on the character counter.
     */
    characterCounter: PropTypes.string,
    /**
     * Style controlling the layout of the counter.
     */
    inline: PropTypes.string,
    /**
     * Style applied to the separator element of the character counter.
     */
    separator: PropTypes.string,
    /**
     * Style applied to the max counter element of the character counter.
     */
    maxCharacter: PropTypes.string,
    /**
     * Style applied to the current counter element of the character counter.
     */
    currentCounter: PropTypes.string,
    /**
     * Style applied to the character counter when it is disabled.
     */
    disabled: PropTypes.string
  }).isRequired,
  /**
   * An Object containing the various text associated with the text area.
   *
   * -inputLabel: the label on top of the input.
   * -placeholder: the placeholder value of the input.
   * -infoText: the default value of the info text below the input.
   * -warningText: the value when a validation fails.
   * -maxCharQuantityWarningText: the message that appears when there are too many characters.
   * -minCharQuantityWarningText: the message that appears when there are too few characters.
   * -requiredWarningText: the message that appears when the input is empty and required.
   */
  inputTextConfiguration: PropTypes.shape({
    inputLabel: PropTypes.string,
    placeholder: PropTypes.string,
    infoText: PropTypes.string,
    warningText: PropTypes.string,
    maxCharQuantityWarningText: PropTypes.string,
    minCharQuantityWarningText: PropTypes.string,
    requiredWarningText: PropTypes.string
  }),
  /**
   * The maximum allowed length of the characters, if this value is null or undefined no check
   * will be performed.
   */
  maxCharQuantity: PropTypes.number,
  /**
   * The number of rows of the text area
   */
  rows: PropTypes.number,
  /**
   * The initial value of the text area.
   */
  value: PropTypes.string,
  /**
   * The function that will be executed onChange, allows modification of the input,
   * it receives the value and must return a value otherwise the input will be empty.
   */
  onChange: PropTypes.func,
  /**
   * If ´true´ the text area is disabled.
   */
  disabled: PropTypes.bool
};

HvTextArea.defaultProps = {
  inputTextConfiguration: {
    inputLabel: "",
    placeholder: "",
    warningText: "",
    maxCharQuantityWarningText: "",
    minCharQuantityWarningText: "",
    requiredWarningText: ""
  },
  rows: 1,
  disabled: false,
  value: "",
  maxCharQuantity: undefined,
  onChange: value => value
};

export default HvTextArea;
