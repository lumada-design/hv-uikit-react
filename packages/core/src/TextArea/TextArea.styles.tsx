import { createClasses } from "@core/utils/classes";

export const { staticClasses, useClasses } = createClasses("HvTextArea", {
  root: { display: "inline-block", width: "100%" },
  disabled: {},
  resizable: { width: "100%" },
  invalid: {},
  baseInput: { clear: "both", float: "left" },
  input: {},
  inputResizable: { width: "100%", resize: "both" },
  labelContainer: { float: "left", display: "flex", alignItems: "flex-start" },
  label: { display: "block", float: "left", paddingBottom: "6px" },
  description: { display: "block", float: "left" },
  characterCounter: {
    display: "block",
    float: "right",
    textAlign: "right",
    marginBottom: "6px",
  },
  error: { float: "left" },
});
