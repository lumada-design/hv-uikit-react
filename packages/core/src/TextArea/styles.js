import { outlineStyles } from "../Focus/styles";

const styles = theme => ({
  input: {
    height: "auto",
    padding: "5px 10px",
    overflow: "auto",
    marginLeft: "0px",
    marginRight: "0px",
    minHeight: "21px",
    minWidth: "150px",
    maxWidth: "610px"
  },
  resize: {
    resize: "auto",
    width: "100%"
  },
  defaultWith: {
    width: "610px"
  },
  inputRoot: {
    borderWidth: " 1px",
    "&:hover": {
      borderWidth: " 1px"
    }
  },
  inputRootDisabled: {
    background: theme.hv.palette.atmosphere.atmo2,
    "&:hover": {
      background: theme.hv.palette.atmosphere.atmo2
    }
  },
  inputRootFocused: {
    borderWidth: " 1px",
    ...outlineStyles,
    "&:hover": {
      borderWidth: " 1px"
    }
  },
  container: {},
  characterCounter: {
    textAlign: "right",
    paddingTop: `${theme.hv.spacing.xs}px`
  },
  inline: {
    display: "inline"
  },
  separator: {
    margin: `0 3px`
  },
  disabled: {
    ...theme.hv.typography.infoText
  },
  maxCharacter: {
    ...theme.hv.typography.infoText
  },
  currentCounter: {
    ...theme.hv.typography.labelText
  },
  textAreaContainer: {
    display: "inline-block"
  }
});

export default styles;
