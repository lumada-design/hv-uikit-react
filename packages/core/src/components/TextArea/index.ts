export { default as textAreaClasses } from "./textAreaClasses";
export * from "./textAreaClasses";
export * from "./TextArea";
