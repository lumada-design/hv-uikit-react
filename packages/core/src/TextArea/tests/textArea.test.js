/* eslint-env jest */

import React from "react";
import { mount, shallow } from "enzyme";

import { HvProvider, HvTextArea } from "../..";
import { Main } from "../stories/TextArea.stories";

describe("TextArea", () => {
  let wrapper;

  beforeEach(async () => {
    wrapper = shallow(
      <HvProvider disableCssBaseline>
        <Main />
      </HvProvider>
    );
  });

  it("should render correctly", () => {
    expect(wrapper.find(HvTextArea)).toMatchSnapshot();
  });
});

describe("TextArea Component", () => {
  let wrapper;

  beforeEach(async () => {
    wrapper = shallow(
      <HvProvider disableCssBaseline>
        <HvTextArea rows={4} />
      </HvProvider>
    );
  });

  it("should render correctly if opened", () => {
    expect(wrapper.find(HvTextArea)).toMatchSnapshot();
  });

  it("should render the count label correctly", () => {
    const wrapperMount = mount(
      <HvProvider disableCssBaseline>
        <HvTextArea rows={4} defaultValue="test" middleCountLabel="of" maxCharQuantity={10} />
      </HvProvider>
    );
    const labelCount = wrapperMount.find("HvTextArea").find("label").at(0).text();
    expect(labelCount).toBe("4");
  });

  it("should render the count label correctly and show the warning", () => {
    const wrapperMount = mount(
      <HvProvider disableCssBaseline>
        <HvTextArea rows={4} defaultValue="tests" middleCountLabel="of" maxCharQuantity={3} />
      </HvProvider>
    );

    const labelCount = wrapperMount.find("HvTextArea").find("label").at(0).text();
    expect(labelCount).toBe("5");
  });

  it("should use the ref passed by the caller", () => {
    const ref = React.createRef(null);
    expect(ref.current).toBe(null);

    mount(
      <HvProvider disableCssBaseline>
        <HvTextArea inputRef={ref} />
      </HvProvider>
    );

    expect(ref.current).toBeInstanceOf(HTMLTextAreaElement);
  });
});
