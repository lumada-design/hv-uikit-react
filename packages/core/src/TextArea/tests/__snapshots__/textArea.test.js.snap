// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea Component should render correctly if opened 1`] = `
<Fragment>
  <WithStyles(HvInput)
    className=""
    classes={
      Object {
        "container": undefined,
        "input": undefined,
      }
    }
    disabled={false}
    iconVisible={false}
    initialValue="test"
    labels={
      Object {
        "inputLabel": "",
        "maxCharQuantityWarningText": "",
        "minCharQuantityWarningText": "",
        "placeholder": "",
        "requiredWarningText": "",
        "warningText": "",
      }
    }
    multiline={true}
    onChange={[Function]}
    rows={4}
    validate={false}
  />
</Fragment>
`;

exports[`TextArea withStyles should render correctly 1`] = `
<MuiThemeProviderOld
  sheetsManager={Map {}}
  theme={
    Object {
      "breakpoints": Object {
        "between": [Function],
        "down": [Function],
        "keys": Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ],
        "only": [Function],
        "up": [Function],
        "values": Object {
          "lg": 1270,
          "md": 960,
          "sm": 600,
          "xl": 1920,
          "xs": 0,
        },
        "width": [Function],
      },
      "direction": "ltr",
      "hv": Object {
        "deprecated": Object {
          "palette": Object {
            "accent": Object {},
            "atmosphere": Object {},
            "semantic": Object {},
            "support": Object {},
          },
          "typography": Object {
            "disabledText": Object {},
          },
          "viz": Object {
            "palette": Object {
              "categorical": Object {},
              "polarizes": Object {},
              "sequential": Object {},
              "undefinedState": Object {},
            },
          },
        },
        "name": "dawn",
        "palette": Object {
          "accent": Object {
            "acce0": "#FFFFFF",
            "acce1": "#414141",
            "acce2": "#146BD2",
            "acce2h": "#4389DB",
            "acce3": "#CC0000",
          },
          "atmosphere": Object {
            "atmo1": "#FFFFFF",
            "atmo2": "#F9F9F9",
            "atmo3": "#F5F5F5",
            "atmo4": "#F0F0F0",
            "atmo5": "#DEDEDE",
            "atmo6": "#BCBCBC",
            "atmo7": "#999999",
          },
          "semantic": Object {
            "sema1": "#63A621",
            "sema10": "#EF9A9A",
            "sema11": "#E57373",
            "sema12": "#EF5350",
            "sema13": "#E53935",
            "sema14": "#C62828",
            "sema15": "#80DEEA",
            "sema16": "#4DD0E1",
            "sema17": "#26C6DA",
            "sema18": "#00ACC1",
            "sema19": "#00838F",
            "sema2": "#72CCCB",
            "sema3": "#E68C17",
            "sema4": "#FF5252",
            "sema5": "#C51162",
            "sema6": "#AA00FF",
            "sema7": "#D3E3F6",
            "sema8": "#DFECD1",
            "sema9": "#F5D8D8",
          },
          "support": Object {
            "supp1": "#0F8B8D",
            "supp2": "#734B6D",
            "supp3": "#4E7599",
            "supp4": "#C19C31",
            "supp5": "#546B6B",
          },
        },
        "spacing": Object {
          "lg": "60",
          "md": "30",
          "sm": "20",
          "xl": "90",
          "xs": "10",
        },
        "type": "light",
        "typography": Object {
          "3xlTitle": Object {
            "color": "#414141",
            "fontSize": "42px",
            "fontWeight": "600",
            "letterSpacing": "0.02em",
            "lineHeight": "52px",
          },
          "4xlTitle": Object {
            "color": "#414141",
            "fontSize": "52px",
            "fontWeight": "400",
            "letterSpacing": "0.02em",
            "lineHeight": "60px",
          },
          "5xlTitle": Object {
            "color": "#414141",
            "fontSize": "52px",
            "fontWeight": "600",
            "letterSpacing": "0.02em",
            "lineHeight": "60px",
          },
          "disabledButtonText": Object {
            "color": "#999999",
            "fontSize": "14px",
            "fontWeight": "600",
            "letterSpacing": "0.02em",
            "lineHeight": "20px",
          },
          "disabledText": Object {
            "color": "#999999",
            "fontSize": "12px",
            "fontWeight": "400",
            "letterSpacing": "0.02em",
            "lineHeight": "16px",
          },
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "highlightText": Object {
            "color": "#414141",
            "fontSize": "14px",
            "fontWeight": "600",
            "letterSpacing": "0.02em",
            "lineHeight": "20px",
          },
          "infoText": Object {
            "color": "#999999",
            "fontSize": "12px",
            "fontWeight": "400",
            "letterSpacing": "0.02em",
            "lineHeight": "16px",
          },
          "inlineLink": Object {
            "color": "#146BD2",
            "fontSize": "14px",
            "fontWeight": "600",
            "letterSpacing": "0.02em",
            "lineHeight": "20px",
          },
          "lTitle": Object {
            "color": "#414141",
            "fontSize": "32px",
            "fontWeight": "400",
            "letterSpacing": "0.02em",
            "lineHeight": "40px",
          },
          "labelText": Object {
            "color": "#414141",
            "fontSize": "12px",
            "fontWeight": "600",
            "letterSpacing": "0.02em",
            "lineHeight": "16px",
          },
          "mTitle": Object {
            "color": "#414141",
            "fontSize": "22px",
            "fontWeight": "600",
            "letterSpacing": "0.02em",
            "lineHeight": "30px",
          },
          "normalText": Object {
            "color": "#414141",
            "fontSize": "14px",
            "fontWeight": "400",
            "letterSpacing": "0.02em",
            "lineHeight": "20px",
          },
          "placeholderText": Object {
            "color": "#999999",
            "fontSize": "14px",
            "fontWeight": "400",
            "letterSpacing": "0.02em",
            "lineHeight": "20px",
          },
          "sLink": Object {
            "color": "#146BD2",
            "fontSize": "12px",
            "fontWeight": "400",
            "letterSpacing": "0.02em",
            "lineHeight": "16px",
          },
          "sText": Object {
            "color": "#414141",
            "fontSize": "12px",
            "fontWeight": "400",
            "letterSpacing": "0.02em",
            "lineHeight": "16px",
          },
          "sTitle": Object {
            "color": "#414141",
            "fontSize": "22px",
            "fontWeight": "400",
            "letterSpacing": "0.02em",
            "lineHeight": "30px",
          },
          "selectedNavText": Object {
            "color": "#CC0000",
            "fontSize": "14px",
            "fontWeight": "600",
            "letterSpacing": "0.02em",
            "lineHeight": "20px",
          },
          "selectedText": Object {
            "background": "#414141",
            "color": "#F9F9F9",
            "fontSize": "14px",
            "fontWeight": "600",
            "letterSpacing": "0.02em",
            "lineHeight": "20px",
          },
          "vizText": Object {
            "color": "#414141",
            "fontSize": "10px",
            "fontWeight": "400",
            "letterSpacing": "0.02em",
            "lineHeight": "15px",
          },
          "xlTitle": Object {
            "color": "#414141",
            "fontSize": "32px",
            "fontWeight": "600",
            "letterSpacing": "0.02em",
            "lineHeight": "40px",
          },
          "xsTitle": Object {
            "color": "#414141",
            "fontSize": "18px",
            "fontWeight": "600",
            "letterSpacing": "0.02em",
            "lineHeight": "28px",
          },
          "xxlTitle": Object {
            "color": "#414141",
            "fontSize": "42px",
            "fontWeight": "400",
            "letterSpacing": "0.02em",
            "lineHeight": "52px",
          },
          "xxsTitle": Object {
            "color": "#414141",
            "fontSize": "18px",
            "fontWeight": "400",
            "letterSpacing": "0.02em",
            "lineHeight": "28px",
          },
        },
        "viz": Object {
          "palette": Object {
            "categorical": Object {
              "cviz1": "#2DB3E0",
              "cviz10": "#419393",
              "cviz11": "#58C9DD",
              "cviz12": "#F8C189",
              "cviz13": "#528BBF",
              "cviz14": "#DE64BD",
              "cviz15": "#73BF9C",
              "cviz16": "#9672C1",
              "cviz17": "#EB95AF",
              "cviz18": "#646A98",
              "cviz19": "#D88136",
              "cviz2": "#FF990D",
              "cviz20": "#309B59",
              "cviz3": "#D31F6B",
              "cviz4": "#FFD00A",
              "cviz5": "#18C38A",
              "cviz6": "#098CB2",
              "cviz7": "#EA5B86",
              "cviz8": "#9B469B",
              "cviz9": "#F4763D",
            },
            "polarizes": Object {
              "cviz21": "#869F1E",
              "cviz22": "#A4991C",
              "cviz23": "#C4931A",
              "cviz24": "#E68C17",
              "cviz25": "#F27034",
              "cviz26": "#FF5252",
              "cviz27": "#EC3D57",
              "cviz28": "#D8265D",
              "sema1": "#63A621",
              "sema5": "#C51162",
            },
            "sequential": Object {
              "cviz1": "#2DB3E0",
              "cviz1_100": "#E3F1F6",
              "cviz1_200": "#CFEAF3",
              "cviz1_300": "#BAE3F3",
              "cviz1_400": "#A6DCEE",
              "cviz1_500": "#92D5EC",
              "cviz1_600": "#7ECEE9",
              "cviz1_700": "#69C7E7",
              "cviz1_800": "#55C0E4",
              "cviz1_900": "#40B9E2",
            },
            "undefinedState": Object {
              "atmo5": "#DEDEDE",
            },
          },
        },
      },
      "mixins": Object {
        "gutters": [Function],
        "toolbar": Object {
          "@media (min-width:0px) and (orientation: landscape)": Object {
            "minHeight": 48,
          },
          "@media (min-width:600px)": Object {
            "minHeight": 64,
          },
          "minHeight": 56,
        },
      },
      "overrides": Object {
        "MuiAppBar": Object {
          "colorDefault": Object {
            "backgroundColor": "#FFFFFF",
            "contrastText": "#414141",
          },
          "root": Object {
            "@media (min-width: 600px)": Object {
              "height": "50px",
            },
            "borderTop": "4px solid #CC0000",
          },
        },
        "MuiToolbar": Object {
          "dense": Object {
            "minHeight": "46px",
          },
          "gutters": Object {
            "@media (min-width:600px)": Object {
              "paddingLeft": "20px",
              "paddingRight": "20px",
            },
            "paddingLeft": "20px",
            "paddingRight": "20px",
          },
          "root": Object {
            "@media (min-width: 600px)": Object {
              "minHeight": "46px",
            },
            "alignItems": "center",
          },
        },
      },
      "palette": Object {
        "action": Object {
          "active": "rgba(0, 0, 0, 0.54)",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "disabledBackground": "rgba(0, 0, 0, 0.12)",
          "hover": "rgba(0, 0, 0, 0.08)",
          "hoverOpacity": 0.08,
          "selected": "rgba(0, 0, 0, 0.14)",
        },
        "augmentColor": [Function],
        "background": Object {
          "default": "#F9F9F9",
          "paper": "#fff",
        },
        "common": Object {
          "black": "#000",
          "white": "#fff",
        },
        "contrastThreshold": 3,
        "divider": "rgba(0, 0, 0, 0.12)",
        "error": Object {
          "contrastText": "#fff",
          "dark": "#d32f2f",
          "light": "#e57373",
          "main": "#f44336",
        },
        "getContrastText": [Function],
        "grey": Object {
          "100": "#f5f5f5",
          "200": "#eeeeee",
          "300": "#e0e0e0",
          "400": "#bdbdbd",
          "50": "#fafafa",
          "500": "#9e9e9e",
          "600": "#757575",
          "700": "#616161",
          "800": "#424242",
          "900": "#212121",
          "A100": "#d5d5d5",
          "A200": "#aaaaaa",
          "A400": "#303030",
          "A700": "#616161",
        },
        "primary": Object {
          "contrastText": "#fff",
          "dark": "rgb(14, 74, 147)",
          "light": "#4389DB",
          "main": "#146BD2",
        },
        "secondary": Object {
          "contrastText": "#fff",
          "dark": "rgb(45, 45, 45)",
          "light": "#999999",
          "main": "#414141",
        },
        "text": Object {
          "disabled": "#999999",
          "hint": "rgba(0, 0, 0, 0.38)",
          "primary": "#414141",
          "secondary": "rgba(0, 0, 0, 0.54)",
        },
        "tonalOffset": 0.2,
        "type": "light",
      },
      "props": Object {
        "MuiButtonBase": Object {
          "disableRipple": true,
        },
        "MuiInput": Object {
          "disableUnderline": true,
        },
      },
      "shadows": Array [
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
      ],
      "shape": Object {
        "borderRadius": 0,
      },
      "spacing": Object {
        "lg": "60",
        "md": "30",
        "sm": "20",
        "unit": 8,
        "xl": "90",
        "xs": "10",
      },
      "transitions": Object {
        "create": [Function],
        "duration": Object {
          "complex": 375,
          "enteringScreen": 225,
          "leavingScreen": 195,
          "short": 250,
          "shorter": 200,
          "shortest": 150,
          "standard": 300,
        },
        "easing": Object {
          "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
          "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
          "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
          "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
        },
        "getAutoHeightDuration": [Function],
      },
      "typography": Object {
        "body1": Object {
          "color": "#414141",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "14px",
          "fontWeight": "400",
          "letterSpacing": "0.02em",
          "lineHeight": "20px",
        },
        "body1Next": Object {
          "color": "#414141",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "1rem",
          "fontWeight": 400,
          "lineHeight": 1.5,
        },
        "body2": Object {
          "color": "#999999",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "12px",
          "fontWeight": "400",
          "letterSpacing": "0.02em",
          "lineHeight": "16px",
        },
        "body2Next": Object {
          "color": "#414141",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "0.875rem",
          "fontWeight": 400,
          "lineHeight": 1.5,
        },
        "button": Object {
          "color": "#414141",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "14px",
          "fontWeight": "600",
          "letterSpacing": "0.02em",
          "lineHeight": "20px",
          "textTransform": "uppercase",
        },
        "buttonNext": Object {
          "color": "#414141",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "0.875rem",
          "fontWeight": 500,
          "lineHeight": 1.75,
          "textTransform": "uppercase",
        },
        "caption": Object {
          "color": "#414141",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "0.75rem",
          "fontWeight": 400,
          "lineHeight": 1.66,
        },
        "captionNext": Object {
          "color": "#414141",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "0.75rem",
          "fontWeight": 400,
          "lineHeight": 1.66,
        },
        "display1": Object {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "2.125rem",
          "fontWeight": 400,
          "lineHeight": "1.20588em",
        },
        "display2": Object {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "2.8125rem",
          "fontWeight": 400,
          "lineHeight": "1.13333em",
          "marginLeft": "-.02em",
        },
        "display3": Object {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "3.5rem",
          "fontWeight": 400,
          "letterSpacing": "-.02em",
          "lineHeight": "1.30357em",
          "marginLeft": "-.02em",
        },
        "display4": Object {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "7rem",
          "fontWeight": 300,
          "letterSpacing": "-.04em",
          "lineHeight": "1.14286em",
          "marginLeft": "-.04em",
        },
        "fontFamily": Array [
          "'Open Sans'",
          "Arial",
          "Helvetica",
          "sans-serif",
        ],
        "fontSize": 14,
        "fontWeightLight": 300,
        "fontWeightMedium": 500,
        "fontWeightRegular": 400,
        "h1": Object {
          "color": "#414141",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "32px",
          "fontWeight": "600",
          "letterSpacing": "0.02em",
          "lineHeight": "40px",
        },
        "h2": Object {
          "color": "#414141",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "32px",
          "fontWeight": "400",
          "letterSpacing": "0.02em",
          "lineHeight": "40px",
        },
        "h3": Object {
          "color": "#414141",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "22px",
          "fontWeight": "600",
          "letterSpacing": "0.02em",
          "lineHeight": "30px",
        },
        "h4": Object {
          "color": "#414141",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "22px",
          "fontWeight": "400",
          "letterSpacing": "0.02em",
          "lineHeight": "30px",
        },
        "h5": Object {
          "color": "#414141",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "1.5rem",
          "fontWeight": 400,
          "lineHeight": 1.33,
        },
        "h6": Object {
          "color": "#414141",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "1.25rem",
          "fontWeight": 500,
          "lineHeight": 1.6,
        },
        "headline": Object {
          "color": "#414141",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "1.5rem",
          "fontWeight": 400,
          "lineHeight": "1.35417em",
        },
        "overline": Object {
          "color": "#414141",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "0.75rem",
          "fontWeight": 400,
          "lineHeight": 2.66,
          "textTransform": "uppercase",
        },
        "pxToRem": [Function],
        "round": [Function],
        "subheading": Object {
          "color": "#414141",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "1rem",
          "fontWeight": 400,
          "lineHeight": "1.5em",
        },
        "subtitle1": Object {
          "color": "#414141",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "14px",
          "fontWeight": "600",
          "letterSpacing": "0.02em",
          "lineHeight": "20px",
        },
        "subtitle2": Object {
          "color": "#414141",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "12px",
          "fontWeight": "600",
          "letterSpacing": "0.02em",
          "lineHeight": "16px",
        },
        "suppressDeprecationWarnings": true,
        "title": Object {
          "color": "#414141",
          "fontFamily": Array [
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
            "'Open Sans'",
            "Arial",
            "Helvetica",
            "sans-serif",
          ],
          "fontSize": "1.3125rem",
          "fontWeight": 500,
          "lineHeight": "1.16667em",
        },
        "useNextVariants": true,
      },
      "zIndex": Object {
        "appBar": 1100,
        "drawer": 1200,
        "mobileStepper": 1000,
        "modal": 1300,
        "snackbar": 1400,
        "tooltip": 1500,
      },
    }
  }
>
  <WithStyles(CssBaseline) />
  <ContextProvider
    value={
      Object {
        "changeTheme": [Function],
        "router": null,
      }
    }
  >
    <WithStyles(HvTextArea) />
  </ContextProvider>
</MuiThemeProviderOld>
`;
